api/capi/src/nnstreamer-capi-single.c

```diff
@@ -442,6 +437,16 @@ ml_single_invoke (ml_single_h single,
     return ML_ERROR_TIMED_OUT;
   }
 
+  /* Allocate output buffer */
+  status = ml_tensors_data_create (&single_h->out_info, output);
+  if (status != ML_ERROR_NONE) {
+    ml_loge ("Failed to allocate the memory block.");
+    *output = NULL;
+    return status;
+  }
+
+  result = (ml_tensors_data_s *) (*output);
```

Comment:
Why was this code block moved?

Reply:
When this function returns error (ML_ERROR_TIMED_OUT in line 437), the output-handle is allocated. So I moved these block after retrieving the buffer from appsink.

Reply:
Then you need
```
gst_sample_unref (sample);
```

Reply:
1. sample unref
Right, I added gst_sample_unref() if failed to allocate mem block.

2. buffer
gst_app_src_push_buffer() takes ownership of the buffer.

Reply:
Ok. LTGM now.

Reply:
add buffer_unref if failed to push buffer to appsrc (if refcount is 0 already, it will do nothing.)